packages/fx-core/src/plugins/resource/frontend/env.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import * as dotenv from "dotenv";
+import fs from "fs-extra";
+import * as path from "path";
+import * as os from "os";
+import { TelemetryHelper } from "./utils/telemetry-helper";
+import { TelemetryEvent } from "./constants";
+
+export interface RemoteEnvs {
+  teamsfxRemoteEnvs: { [key: string]: string };
+  customizedRemoteEnvs: { [name: string]: string };
+}
+
+export const envFileName = (envName: string): string => `env.teamsfx.${envName}`;
+
+export const EnvKeys = Object.freeze({
+  FuncEndpoint: "REACT_APP_FUNC_ENDPOINT",
+  FuncName: "REACT_APP_FUNC_NAME",
+  RuntimeEndpoint: "REACT_APP_TEAMSFX_ENDPOINT",
+  StartLoginPage: "REACT_APP_START_LOGIN_PAGE_URL",
+  ClientID: "REACT_APP_CLIENT_ID",
+});
+
+export function initEnvs(): RemoteEnvs {
+  const result: RemoteEnvs = {
+    teamsfxRemoteEnvs: {},
+    customizedRemoteEnvs: {},
+  };
+
+  return result;
+}
+
+export async function loadEnvFile(envName: string, folder: string): Promise<RemoteEnvs> {
+  const envPath = path.join(folder, envFileName(envName));
+  if (!(await fs.pathExists(envPath))) {
+    return initEnvs();
+  }
+  const envs = dotenv.parse(await fs.readFile(envPath));
+  const result = initEnvs();
+  const entries = Object.entries(envs);
+  for (const [key, value] of entries) {
+    if (Object.values(EnvKeys).includes(key)) {
+      result.teamsfxRemoteEnvs[key] = value;
+    } else {
+      result.customizedRemoteEnvs[key] = value;
+    }
+  }
+  return result;
+}
+
+export async function saveEnvFile(
+  envName: string,
+  folder: string,
+  envs: RemoteEnvs
+): Promise<void> {
+  try {
+    const configs = await loadEnvFile(envName, folder);
+    const newConfigs: RemoteEnvs = {
+      teamsfxRemoteEnvs: { ...configs.teamsfxRemoteEnvs, ...envs.teamsfxRemoteEnvs },
+      customizedRemoteEnvs: { ...configs.customizedRemoteEnvs, ...envs.customizedRemoteEnvs },
+    };
+
+    if (JSON.stringify(newConfigs) === JSON.stringify(configs)) {
```

Comment:
There's only one case that the order would be different: user change the order of those configs generated by toolkit. 
In that case, I think it's reasonable to update the env file because user modifies it. Even though the modification doesn't take effect.
In case the order changed, then we just update the env file and skip deployment feature may not work as expected. I think it's ok because everything work true.